TRRandomizerCore/Editors/TR2RandoEditor.cs

```diff
@@ -31,15 +31,81 @@ protected override void StoreConfig(Config config)
 
     protected override int GetSaveTarget(int numLevels)
     {
-        int target = base.GetSaveTarget(numLevels) + Settings.GetSaveTarget(numLevels);
-        if (Settings.RandomizeItems && Settings.RandomizeItemSprites)
+        int target = base.GetSaveTarget(numLevels);
+
+        if (Settings.RandomizeGameStrings || Settings.ReassignPuzzleNames)
+        {
+            target++;
+        }
+
+        if (Settings.RandomizeNightMode)
+        {
+            target += numLevels;
+            if (!Settings.RandomizeTextures)
+            {
+                // Texture randomizer will run if night mode is on to ensure skyboxes and such like match
+                target += numLevels;
+            }
+        }
+
+        if (Settings.RandomizeSecrets)
+        {
+            target += numLevels;
+        }
+
+        if (Settings.RandomizeAudio)
+        {
+            target += numLevels;
+        }
+
+        if (Settings.RandomizeItems)
+        {
+            // Standard/key item rando followed by unarmed logic after enemy rando
+            target += numLevels * 2;
+            if (Settings.RandomizeItemSprites)
+            {
+                target += numLevels;
+            }
+            if (Settings.IncludeKeyItems)
+            {
+                target += numLevels;
+            }
+        }
+
+        if (Settings.RandomizeStartPosition)
         {
             target += numLevels;
         }
+
+        if (Settings.DeduplicateTextures)
+        {
+            // *2 because of multithreaded approach
+            target += numLevels * 2;
+        }
+
         if (Settings.RandomizeEnemies)
         {
-            target += numLevels; // Used to eliminate unused enemies prior to any other processing
+            // *4 => 3 for multithreading work, 1 for ModelAdjuster
+            target += Settings.CrossLevelEnemies ? numLevels * 4 : numLevels;
+            // And again for eliminating unused enemies
+            target += numLevels;
         }
+
+        if (Settings.RandomizeTextures)
+        {
+            // *3 because of multithreaded approach
+            target += numLevels * 3;
+        }
+
+        if (Settings.RandomizeOutfits)
+        {
+            // *2 because of multithreaded approach
+            target += numLevels * 2;
+        }
```

Comment:
Can you elaborate? This sounds like a red flag

Reply:
There are a few randomizer classes that split the work out into threads - mainly ones doing model import work like `TR2EnemyRandomizer`. It's quite clunky at the moment tbh. We trigger progress at different stages so that the progress bar is seen to be shifting - some import work can take a long time due to limits being hit, with it then rolling back and trying again (e.g. with a different enemy import set). So with the threading approach the other levels in this case can still make some progress and the user isn't left thinking it's frozen.

This particular bit of code is moved from `RandomizerSettings` into this class. It was originally put there as the idea was that all of the editors would use the same approach, but there are differences between the games as things have evolved, so I don't think it's viable currently.

I'd like to get rid of the manual threading approach and modernize it all with tasks eventually. Then I think we could simplify the progress target calculations.

Reply:
Yeah, what if we let each task report its progress individually? IMO it's OK to have multiple progress bars, too. Eg randomizing enemies: 55% randomizing floordata: 80% 
Probably doesn't belong to this PR tho.

Reply:
Agreed, I'll create a follow up issue for this. I think it could even help boost performance when we switch to tasks - currently we limit to 3 threads, so hopefully letting the system manage it all in that respect will pay off for us.